tests/longhaul/base_object_test.go

```diff
@@ -0,0 +1,105 @@
+package longhaul
+
+import (
+	"math/rand"
+	"sync"
+	"testing"
+
+	"github.com/icrowley/fake"
+	"github.com/rook/rook/tests/framework/clients"
+	"github.com/rook/rook/tests/framework/installer"
+	"github.com/rook/rook/tests/framework/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+// Set up rook cluster if necessary
+// create object store if necessary
+// create object store user if necessary
+// returns k8s helper, install helper and s3 helper for the object store that was just created.
+func setUpRook(t func() *testing.T, namespace string) (*utils.K8sHelper, *installer.InstallHelper, *clients.TestClient) {
+	kh, err := utils.CreateK8sHelper(t)
+	assert.Nil(t(), err)
+
+	i := installer.NewK8sRookhelper(kh.Clientset, t)
+	if !kh.IsRookInstalled(namespace) {
+		isRookInstalled, err := i.InstallRookOnK8sWithHostPathAndDevices(namespace, "bluestore", "/temp/rookBackup", true, 3)
+		require.NoError(t(), err)
+		require.True(t(), isRookInstalled)
+	}
+	helper, err := clients.CreateTestClient(kh, namespace)
+	if err != nil {
+		logger.Errorf("Cannot create rook test client, er -> %v", err)
+		t().FailNow()
+	}
+
+	return kh, i, helper
+}
+
+func createObjectStoreAndUser(t func() *testing.T, kh *utils.K8sHelper, tc *clients.TestClient, namespace string, storeName string, userId string, userName string) *utils.S3Helper {
```

Comment:
could there be a helper factored out from the smoke tests for this?

Reply:
we have only Object store test currently(smoke test). This is a helper specific for load test, if future as we write more tests this will be refactored.

Reply:
what is the difference between the smoke test and this test? Does there need to be a difference?

Reply:
Smoke Test is functional, we set up only one object store  and do a lot of checks. Load test does fewer checks but sets up multiple object store. eg. Smoke test does a check for deleting users and buckets. But load test doesnt. It just sets up object store and uses it for a long time.